daemon/oci_linux.go

```diff
@@ -988,6 +986,10 @@ func WithResources(c *container.Container) coci.SpecOpts {
 			},
 			Pids: getPidsLimit(r),
 		}
+		if r.BlkioWeight != 0 {
+			w := r.BlkioWeight
+			specResources.BlockIO.Weight = &w
+		}
 
 		if s.Linux.Resources != nil && len(s.Linux.Resources.Devices) > 0 {
 			specResources.Devices = s.Linux.Resources.Devices
```

Comment:
Looking at this, I feel like `WithResources()` is doing too much, or we should split out a `WithBlkIO()` - it looks like this part is here, to prevent `WithResources()` from overwriting `Devices` (which probably is set separately), so trying to merge multiple options.

Reply:
No, ensuring that mutations to the spec by `WithDevices()` are not clobbered is _not_ `WithResources()` doing too much. It's just `WithResources()` being a good(-ish) implementation of a functional option, and what allows the functional options pattern to be used to factor distinct concerns into distinct option functions. Factoring out a `WithBlkIO()` function would also have to ensure that it does not clobber changes to the spec applied by `WithResources()` and other options.

That all being said, the way that `WithResources()` goes about not clobbering existing `LinuxResources` fields it does not otherwise touch is not particularly forward-compatible as it will zero out all other fields aside from `Devices`. I'll fix that.